data/rulesets.ts

```diff
@@ -1357,4 +1357,13 @@ export const Rulesets: {[k: string]: FormatData} = {
 		desc: "Bans move combinations on Pok\u00e9mon that weren't legal in Japanese versions of Gen 1.",
 		// Implemented in mods/gen1jpn/rulesets.ts
 	},
+	noswitch: {
+		effectType: 'Rule',
+		name: 'No Switch',
+		desc: 'Players cannot switch their Pok&eacute;mon, except with an item, move, or Ability.',
+		onTrapPokemon(pokemon) {
+			pokemon.trapped = true;
+			this.add('rule', '-activate', pokemon, 'trapped');
```

Comment:
Uh this is going to look super ugly and also be very spammy (appearing for every pokemon every turn)

Reply:
Done, thx.